src/codecs/webp/extended.rs

```diff
@@ -0,0 +1,683 @@
+use std::convert::TryInto;
+use std::io::{self, Error, Read};
+use std::{error, fmt};
+
+use super::decoder::{read_chunk, WebPRiffChunk};
+use super::lossless::{LosslessDecoder, LosslessFrame};
+use super::vp8::{Frame as VP8Frame, Vp8Decoder};
+use crate::error::DecodingError;
+use crate::image::ImageFormat;
+use crate::{color, Delay, Frame, Frames, ImageError, ImageResult, Rgba, RgbaImage};
+use byteorder::{LittleEndian, ReadBytesExt};
+
+//all errors that can occur while parsing extended chunks in a WebP file
+#[derive(Debug, Clone, Copy)]
+enum DecoderError {
+    ChunkInvalid,
+    SizeMismatch,
```

Comment:
Can you split this into multiple variants as well, depending on how that mismatch occurred.